### PR TITLE
Dump discarders

### PR DESCRIPTION
**Pull request by modernplumbing**
Dumps the contents of the `discarder_stats_fb`, `discarder_stats_bb`, and `inode_discarders` maps.

### Reviewer's Checklist

- [x] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [x] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [x] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: pkg/security/probe/probe.go [hunk 1]
```diff
@@ -978,6 +978,78 @@ func (p *Probe) SelectProbes(rs *rules.RuleSet) error {
...
+	} else {
+		var fbStat uint32
+		for entries := statsFbMap.Iterate(); entries.Next(&fbStat, &dStats); {
+			fmt.Fprintf(dump, "%d: %+v\n", fbStat, dStats)
+		}
+	}
+
+	fmt.Fprintf(dump, "\nDiscarder Stats BB\n")
+	statsBbMap, _, statsBBErr := p.manager.GetMap("discarder_stats_bb")
+	if err != nil {
```
Comment (safchain) [inline pkg/security/probe/probe.go:1033]: is it not `statsBBErr` here ?

File: pkg/security/probe/probe.go [hunk 2]
```diff
@@ -978,6 +978,78 @@ func (p *Probe) SelectProbes(rs *rules.RuleSet) error {
...
+		return "", err
+	}
+
+	fmt.Fprintf(dump, "Discarder Dump\n%s\n", time.Now().UTC().String())
+	fmt.Fprintf(dump, "\nInode Discarders\n")
+
+	discardedInodeCount := 0
+	var inode inodeDiscarderMapEntry
+	var inodeParams inodeDiscarderParams
+	for entries := p.inodeDiscarders.Iterate(); entries.Next(&inode, &inodeParams); {
```
Comment (safchain) [inline pkg/security/probe/probe.go:1001]: there is a risk here of having the map modified during the iteration which can break the iteration. In order to ensure that we won't fall into an infinite loop I would check that the `discarderInodeCount` is always lower than the map size. what do you think ?

File: pkg/security/probe/probe.go [hunk 3]
```diff
@@ -978,6 +978,78 @@ func (p *Probe) SelectProbes(rs *rules.RuleSet) error {
...
+		fmt.Fprintf(dump, "%d: %+v\n", discardedInodeCount, inode)
+		fmt.Fprintf(dump, "%d: %+v\n", discardedInodeCount, inodeParams)
+	}
+
+	fmt.Fprintf(dump, "\nProcess Discarders\n")
+
+	discardedPIDCount := 0
+	var pid uint32
+	var pidParams pidDiscarderParams
+	for entries := p.pidDiscarders.Iterate(); entries.Next(&pid, &pidParams); {
```
Comment (safchain) [inline pkg/security/probe/probe.go:1012]: same here

**Review by safchain — COMMENTED** [review]
Just one tiny concern otherwise LGTM

File: pkg/security/probe/probe.go (hunk 1 above)
Comment (modernplumbing) [reply pkg/security/probe/probe.go:1033]: yes! thank you

File: pkg/security/probe/probe.go (hunk 2 above)
Comment (paulcacheux) [reply pkg/security/probe/probe.go:1001]: There is no easy way of counting the actual number of elements in the map, do you mean the "max_entries" ?

File: cmd/security-agent/app/runtime.go [hunk 4]
```diff
@@ -994,3 +1010,21 @@ func downloadPolicy(cmd *cobra.Command, args []string) error {
 	_, err = outputWriter.Write(resBytes)
 	return err
 }
+
+func dumpDiscarders(cmd *cobra.Command, args []string) error {
+	runtimeSecurityClient, err := secagent.NewRuntimeSecurityClient()
+	if err != nil {
+		return fmt.Errorf("unable to create a runtime security client instance: %")
```
Comment (paulcacheux) [inline cmd/security-agent/app/runtime.go:1017]:
```suggestion
		return fmt.Errorf("unable to create a runtime security client instance: %w")
```

**Review by paulcacheux — APPROVED** [review]

**Pull request merged**